### ISSUE TITLE
Fancier events? (one time events and individualized off)

### ISSUE THREAD

**Issue by Olliebrown** [alemangui/pizzicato#128]
Just a feature suggestion.  It would be nice to be able to specify an event callback to fire only once (pretty typical pattern in JS event programming).  Would usually look like this:

- function `once()` with same signature as `on()`
- Callbacks added with `once()` remove themselves after they are triggered

Another nice feature would be if 'off()' could remove a more specific callback.  Right now if you say `off('end')` it removes ALL callbacks registered with the end event. The API for this one is not as clear as the once() example but here's one idea:
- `off()` would optionally accept the symbol for the callback as a second argument
- When an callback is provided it looks for a match and removes that callback only (if it is found)
- When no callback is provided (the default) it removes all callbacks associated with the event

I've already run into a use case where I need one of these two features to do what I want so I might give this a shot and send a PR if you'd like.

**Comment by alemangui** [comment]
Hi @Olliebrown,

Thanks for the suggestion. I think the ```once``` would be a good idea if it works like Jquery's [one](https://api.jquery.com/one/) method.

As for the symbol we could pass on to `off`, just to make sure we are on the same page, it would be similar to the ```selector``` argument on Jquery's ```off(events [, selector ] )``` correct?

**Comment by Olliebrown** [comment]
Absolutely!  The jQuery approach would be a nice way to do it.  Thanks!

For others that may wind up here, I worked around this for now with the following very hackish code.  This assumes that no other 'end' callbacks are getting added anywhere else so I can be sure it is always the 'last' event in the list:
```JS
sound.on('end', () => {
  playingCount--
  if (playingCount === 0) {
    setIsPlaying(false)
  }

  // Event should remove itself so it only fires once
  sound._events.end.splice(sound._events.end.length - 1)
})
```
